airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/retriers/default_retrier.py

```diff
@@ -2,16 +2,63 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
-from typing import Optional, Union
+from typing import List, Optional, Union
 
 import requests
-from airbyte_cdk.sources.declarative.requesters.retriers.retrier import Retrier
+from airbyte_cdk.sources.declarative.requesters.retriers.retrier import NonRetriableBehavior, Retrier, RetryBehavior
+
+
+class HttpResponseFilter:
+    def __init__(self, http_codes, error_message_contains=None, predicate=None):
+        if isinstance(http_codes, int):
+            http_codes = set([http_codes])
+        self.http_codes = http_codes
+
+    def matches(self, response) -> bool:
+        # this is only a partial implementation. Also need to check the other fields...
+        if response.status_code in self.http_codes:
+            return True
+        return False
+
+
+DEFAULT_RETRIES = HttpResponseFilter(set(range(500, 600)))
+DEFAULT_RETRIES.http_codes.add(429)
+
+
+class BackoffStrategy:
+    pass
+
+
+class ConstantBackoff:
+    def __init__(self, backoff_in_seconds):
+        self._backoff_in_seconds = backoff_in_seconds
+
+    def backoff(self, response):
+        return self._backoff_in_seconds
 
 
 class DefaultRetrier(Retrier):
-    def __init__(self, max_retries: Optional[int] = 5, retry_factor: float = 5):
+    def __init__(
+        self,
+        max_retries: Optional[int] = 5,
+        retry_factor: float = 5,
```

Comment:
custom exponential backoff, eg with a custom retry_factor, can be implemented by setting `retry_factor` and leaving `backoff` to `None`